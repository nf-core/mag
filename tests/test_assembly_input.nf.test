nextflow_pipeline {

    name "Test pipeline with test_assembly_input profile"
    script "../main.nf"
    tag "pipeline"
    tag "test_assembly_input"
    profile "test_assembly_input"

    test("-profile assembly_input") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            // Output dir: Assembly
            def stable_name_assembly = getAllFilesFromDir(
                params.outdir, include: ['Assembly/**'], relative: true, includeDir: false
            )
            def bowtie2_logs = getAllFilesFromDir(
                params.outdir, include: ['Assembly/{SPAdes,MEGAHIT}/QC/*/*.bowtie2.log']
            )

            // Output directory: GenomeBinning
            def stable_name_binning = getAllFilesFromDir(
                params.outdir, include: ['GenomeBinning/**'], relative: true, includeDir: false
            )
            def stable_path_binning = getAllFilesFromDir(
                params.outdir, include: ['GenomeBinning/**'], ignoreFile: 'tests/.nftignore'
            )
            def concoct_logs = getAllFilesFromDir(
                params.outdir, include: ['GenomeBinning/CONCOCT/stats/*_log.txt']
            )
            def busco_logs = getAllFilesFromDir(
                params.outdir, include: ['GenomeBinning/QC/BUSCO/*/*-busco.log']
            )


            // Output directory: multiqc
            def stable_name_multiqc = getAllFilesFromDir(
                params.outdir, include: ['multiqc/**'], relative: true, includeDir: false
            )
            def stable_path_multiqc = getAllFilesFromDir(
                params.outdir, include: ['multiqc/**'], ignoreFile: 'tests/.nftignore'
            )

            assert workflow.success
            assertAll(
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_mag_software_mqc_versions.yml"),
                ).match() },
                { assert snapshot(stable_name_assembly).match('assembly') },
                { assert snapshot(stable_name_binning,  stable_path_binning ).match('binning') },
                { assert snapshot(stable_name_multiqc, stable_path_multiqc).match('multiqc') },
                // Log checks
                { assert bowtie2_logs.collect {
                        log -> log.readLines().last() ==~ /[0-9.]+% overall alignment rate/
                    }.every()
                },
                { assert concoct_logs.collect {
                        log -> log.readLines().last().contains('CONCOCT Finished')
                    }.every()
                },
                { assert busco_logs.collect {
                        log -> log.readLines().any { it.contains('BUSCO analysis done.') }
                    }.every()
                },
            )
        }
    }
}
