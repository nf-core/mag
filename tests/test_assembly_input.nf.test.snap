{
    "-profile assembly_input": {
        "content": [
            193,
            {
                "ADJUST_MAXBIN2_EXT": {
                    "coreutils": 9.5
                },
                "BIN_SUMMARY": {
                    "python": "3.10.6",
                    "pandas": "1.4.3"
                },
                "BOWTIE2_ASSEMBLY_ALIGN": {
                    "bowtie2": "2.4.2",
                    "samtools": 1.11,
                    "pigz": "2.3.4"
                },
                "BOWTIE2_ASSEMBLY_BUILD": {
                    "bowtie2": "2.4.2"
                },
                "BUSCO_BUSCO": {
                    "busco": "6.0.0"
                },
                "BUSCO_UNTAR": {
                    "untar": 1.34
                },
                "CONCAT_BINQC_TSV": {
                    "csvtk": "0.31.0"
                },
                "CONCOCT_CONCOCT": {
                    "concoct": "1.1.0"
                },
                "CONCOCT_CONCOCTCOVERAGETABLE": {
                    "concoct": "1.1.0"
                },
                "CONCOCT_CUTUPFASTA": {
                    "concoct": "1.1.0"
                },
                "CONCOCT_EXTRACTFASTABINS": {
                    "concoct": "1.1.0"
                },
                "CONCOCT_MERGECUTUPCLUSTERING": {
                    "concoct": "1.1.0"
                },
                "CONVERT_DEPTHS": {
                    "bioawk": 20110810
                },
                "DASTOOL_DASTOOL": {
                    "dastool": "1.1.7"
                },
                "DASTOOL_FASTATOCONTIG2BIN_CONCOCT": {
                    "dastool": "1.1.7"
                },
                "DASTOOL_FASTATOCONTIG2BIN_MAXBIN2": {
                    "dastool": "1.1.7"
                },
                "DASTOOL_FASTATOCONTIG2BIN_METABAT2": {
                    "dastool": "1.1.7"
                },
                "GUNZIP_ASSEMBLYINPUT": {
                    "gunzip": 1.13
                },
                "GUNZIP_BINS": {
                    "gunzip": 1.13
                },
                "MAG_DEPTHS": {
                    "python": "3.6.7",
                    "pandas": "1.1.5"
                },
                "MAG_DEPTHS_PLOT": {
                    "python": "3.9.6",
                    "pandas": "1.3.0",
                    "seaborn": "0.11.0"
                },
                "MAG_DEPTHS_SUMMARY": {
                    "python": "3.10.6",
                    "pandas": "1.4.3"
                },
                "MAXBIN2": {
                    "maxbin2": "2.2.7"
                },
                "METABAT2_JGISUMMARIZEBAMCONTIGDEPTHS_SHORTREAD": {
                    "metabat2": 2.15
                },
                "METABAT2_METABAT2": {
                    "metabat2": 2.17
                },
                "METAEUK_EASYPREDICT": {
                    "metaeuk": "6.a5d39d9"
                },
                "MMSEQS_DATABASES": {
                    "mmseqs": "17.b804f"
                },
                "RENAME_POSTDASTOOL": {
                    "coreutils": 9.5
                },
                "RENAME_PREDASTOOL": {
                    "coreutils": 9.5
                },
                "SEQKIT_STATS": {
                    "seqkit": "2.9.0"
                },
                "SPLIT_FASTA": {
                    "python": "3.6.7",
                    "biopython": "1.7.4",
                    "pandas": "1.1.5"
                },
                "Workflow": {
                    "nf-core/mag": "v5.1.0"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-21T21:09:48.262952969"
    },
    "multiqc": {
        "content": [
            [
                "multiqc/multiqc_data/bowtie2_pe_plot.yaml",
                "multiqc/multiqc_data/busco_plot_bacteria_odb10.yaml",
                "multiqc/multiqc_data/llms-full.txt",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc.parquet",
                "multiqc/multiqc_data/multiqc_bowtie2_bowtie2-2.yaml",
                "multiqc/multiqc_data/multiqc_busco.yaml",
                "multiqc/multiqc_data/multiqc_citations.yaml",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_general_stats.yaml",
                "multiqc/multiqc_data/multiqc_software_versions.yaml",
                "multiqc/multiqc_data/multiqc_sources.yaml",
                "multiqc/multiqc_plots/pdf/bowtie2_pe_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/bowtie2_pe_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/busco_plot_bacteria_odb10-cnt.pdf",
                "multiqc/multiqc_plots/pdf/busco_plot_bacteria_odb10-pct.pdf",
                "multiqc/multiqc_plots/png/bowtie2_pe_plot-cnt.png",
                "multiqc/multiqc_plots/png/bowtie2_pe_plot-pct.png",
                "multiqc/multiqc_plots/png/busco_plot_bacteria_odb10-cnt.png",
                "multiqc/multiqc_plots/png/busco_plot_bacteria_odb10-pct.png",
                "multiqc/multiqc_plots/svg/bowtie2_pe_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/bowtie2_pe_plot-pct.svg",
                "multiqc/multiqc_plots/svg/busco_plot_bacteria_odb10-cnt.svg",
                "multiqc/multiqc_plots/svg/busco_plot_bacteria_odb10-pct.svg",
                "multiqc/multiqc_report.html"
            ],
            [
                "bowtie2_pe_plot.yaml:md5,d47c21d3c14102edf43e36521aa3fcf6",
                "busco_plot_bacteria_odb10.yaml:md5,7f5685578d237300ffde11c05693f754",
                "multiqc_bowtie2_bowtie2-2.yaml:md5,34efd9a91ede5c3d7150b8de2f4c20f0",
                "multiqc_busco.yaml:md5,bd57c1b4856b5536bf89d54b2d8cd807",
                "multiqc_citations.yaml:md5,4aaa7714bb3dba1dd50f383c2ae4f4bb",
                "multiqc_general_stats.yaml:md5,72e40481d31afb109adb0908b18d8758"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-17T15:35:03.829622799"
    }
}